Engine/Server/Src/GameEngine/Application/Application.cpp

```diff
@@ -7,9 +7,41 @@
 
 #include "Application.hpp"
 
-void GameEngine::Application::_packetHandler(Network::UDPPacket packet, const asio::ip::udp::endpoint& endpoint)
+void GameEngine::Application::_packetHandler(Network::UDPPacket packet, const asio::ip::udp::endpoint& endpoint, ECS::Registry& reg)
 {
-    std::cout << "Messages received from " << endpoint << ": [" << "..." << "]" << std::endl;
+    int idxPlayerPacket = -1;
+
+    ECS::SparseArray<IComponent> PlayerComponentArray = reg.get_components<IComponent>("PlayerComponent");
+    for (int index = 0; index < PlayerComponentArray.size(); index++) {
+        PlayerComponent* player = dynamic_cast<PlayerComponent*>(PlayerComponentArray[index].get());
+        if (player->token == packet.getToken()) {
+            idxPlayerPacket = index;
+            break;
+        }
+    }
+    if (idxPlayerPacket == -1) {
+        std::cout << "Message receive from a client no registered" << std::endl;
+        return;
+    }
+
+    if (packet.getMessageType() == static_cast<uint8_t>(MessageType::ArrowKey)) {
+        // TODO: ArrowKey Handler
+    } else if (packet.getMessageType() == static_cast<uint8_t>(MessageType::AlphaKey)) {
+        if (packet.getPayload()[0] == static_cast<uint8_t>(0x04))
+            _sceneManager->processInput(KEY_D, idxPlayerPacket);
+        if (packet.getPayload()[0] == static_cast<uint8_t>(0x11))
+            _sceneManager->processInput(KEY_Q, idxPlayerPacket);
+        if (packet.getPayload()[0] == static_cast<uint8_t>(0x1A))
+            _sceneManager->processInput(KEY_Z, idxPlayerPacket);
+        if (packet.getPayload()[0] == static_cast<uint8_t>(0x13))
+            _sceneManager->processInput(KEY_S, idxPlayerPacket);
+    } else if (packet.getMessageType() == static_cast<uint8_t>(MessageType::NumberKey)) {
+        // TODO: NumberKey Handler
+    } else if (packet.getMessageType() == static_cast<uint8_t>(MessageType::SpecialKey)) {
+        // TODO: SpecialKey Handler
+    } else {
```

Comment:
Firstly, The five main `if`s should be a map because the conditions are the same.

In a second part, all the conditions in the second `if` should also be a map since they got the same condition too